applications/IgaApplication/custom_elements/shell_3p_element.h

```diff
@@ -0,0 +1,460 @@
+        Matrix B22;
+        Matrix B12;
+
+        /**
+        * The default constructor
+        * @param StrainSize: The size of the strain vector in Voigt notation
+        */
+        SecondVariations(const int& mat_size)
+        {
+            B11 = ZeroMatrix(mat_size, mat_size);
+            B22 = ZeroMatrix(mat_size, mat_size);
+            B12 = ZeroMatrix(mat_size, mat_size);
+        }
+    };
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of Shell3pElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(Shell3pElement);
+
+    /// Size types
+    typedef std::size_t SizeType;
+    typedef std::size_t IndexType;
+
+    // GometryType
+    typedef Geometry<Node<3>> GeometryType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor using an array of nodes
+    Shell3pElement(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+        : Element(NewId, pGeometry)
+    {};
+
+    /// Constructor using an array of nodes with properties
+    Shell3pElement(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+        : Element(NewId, pGeometry, pProperties)
+    {};
+
+    /// Default constructor necessary for serialization
+    Shell3pElement()
+        : Element()
+    {};
+
+    /// Destructor.
+    virtual ~Shell3pElement() override
+    {};
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Create with Id, pointer to geometry and pointer to property
+    Element::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties
+    ) const override
+    {
+        return Kratos::make_intrusive<Shell3pElement>(
+            NewId, pGeom, pProperties);
+    };
+
+    /// Create with Id, pointer to geometry and pointer to property
+    Element::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties
+    ) const override
+    {
+        return Kratos::make_intrusive< Shell3pElement >(
+            NewId, GetGeometry().Create(ThisNodes), pProperties);
+    };
+
+    ///@}
+    ///@name Obligatory KRATOS Operations
+    ///@{
+
+    /**
+    * @brief This is called during the assembling process in order
+    *        to calculate the condition right hand side matrix
+    * @param rLeftHandSideMatrix the condition right hand side matrix
+    * @param rCurrentProcessInfo the current process info
+    */
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo) override
+    {
+        const SizeType number_of_nodes = GetGeometry().size();
+        const SizeType mat_size = number_of_nodes * 3;
+
+        if (rRightHandSideVector.size() != mat_size)
+            rRightHandSideVector.resize(mat_size);
+        rRightHandSideVector = ZeroVector(mat_size);
+
+        MatrixType left_hand_side_matrix = Matrix(0, 0);
```

Comment:
```suggestion
        MatrixType left_hand_side_matrix;
```
I think the assignment is not necessary